pkg/destroy/aws/ec2helpers.go

```diff
@@ -276,18 +276,6 @@ func terminateEC2InstanceByInstance(ctx context.Context, ec2Client *ec2.EC2, iam
 		return nil
 	}
 
-	if instance.IamInstanceProfile != nil {
```

Comment:
I see the destroy covers IAM Roles, but I do not see Profiles. Is the IAM Users the same as the Profiles?

Reply:
No. In case of BYO IAM profiles, they should not be destroyed. There is a different call during destroy to remove the `shared` tag from them.
If not BYO profile, then they'll either be discovered via the `owned` tag or via the `UntaggedResources`.

Reply:
The destroy itself happens here: https://github.com/openshift/installer/blob/master/pkg/destroy/aws/iamhelpers.go#L186

Reply:
And here: https://github.com/openshift/installer/blob/master/pkg/destroy/aws/aws.go#L238-L241 which you've worked on before.